packages/normalize-package-json/src/package.js

```diff
@@ -0,0 +1,157 @@
+import { resolve } from 'path'
+import { existsSync, readJSONSync, writeFile } from 'fs-extra'
+import sortPackageJson from 'sort-package-json'
+import consola from 'consola'
+import _ from 'lodash'
+
+const DEFAULTS = {
+  sortDependencies: false,
+  rootDir: process.cwd(),
+  pkgPath: 'package.json',
+  configPath: 'package.js',
+  distDir: 'dist',
+  build: false,
+  suffix: process.env.PACKAGE_SUFFIX ? `-${process.env.PACKAGE_SUFFIX}` : '',
+  hooks: {}
+}
+
+const sortObjectKeys = obj =>
+  _(obj)
+    .toPairs()
+    .sortBy(0)
+    .fromPairs()
+    .value()
+
+export class Package {
+  constructor (options) {
+    const rootDir = process.cwd.call(null)
+    if (rootDir !== DEFAULTS.rootDir) {
+      // Testing? Plz.
+      DEFAULTS.rootDir = rootDir
+    }
+
+    // Assign options
+    this.options = Object.assign({}, DEFAULTS, options)
+
+    // Basic logger
+    this.logger = consola
+
+    // Init (sync)
+    this._init()
+  }
+
+  _init () {
+    // Try to read package.json
+    this.readPkg()
+
+    // Use tagged logger
+    this.logger = consola.withTag(this.pkg.name)
+
+    // Try to load config
+    this.loadConfig()
+  }
+
+  resolvePath (...args) {
+    return resolve(this.options.rootDir, ...args)
+  }
+
+  readPkg () {
+    this.pkg = readJSONSync(this.resolvePath(this.options.pkgPath))
+  }
+
+  loadConfig () {
+    const configPath = this.resolvePath(this.options.configPath)
+
+    if (existsSync(configPath)) {
+      let config = require(configPath)
+      config = config.default || config
+
+      Object.assign(this.options, config)
+    }
+  }
+
+  async callHook (name, ...args) {
+    let fns = this.options.hooks[name]
+
+    if (!fns) {
+      return
+    }
+
+    if (!Array.isArray(fns)) {
+      fns = [fns]
+    }
+
+    for (const fn of fns) {
+      await fn(this, ...args)
+    }
+  }
+
+  load (relativePath, opts) {
+    return new Package(
+      Object.assign(
+        {
+          rootDir: this.resolvePath(relativePath)
+        },
+        opts
+      )
+    )
+  }
+
+  async writePackage () {
+    if (this.options.sortDependencies) {
+      this.sortDependencies()
```

Comment:
I'd bet that this isn't needed, given that most (all?) package managers will do this for you on install.

Reply:
That’s likely something they do in their own monorepo management.

Reply:
@shellscape Yep. But this is for linting our code to make package.json files consistent. And certainly can be optional to disable :)

Reply:
sort-package-json already sort all dependencies.  Would sort-package-json run *before* syncing version between monorepo packages?